hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/TestBlockDeletingService.java

```diff
@@ -369,34 +352,27 @@ public void testContainerThrottle() throws Exception {
     // Process 1 container per interval
     conf.set(
         ScmConfigKeys.OZONE_SCM_KEY_VALUE_CONTAINER_DELETION_CHOOSING_POLICY,
-        RandomContainerDeletionChoosingPolicy.class.getName());
+        TopNOrderedContainerDeletionChoosingPolicy.class.getName());
     conf.setInt(OZONE_BLOCK_DELETING_CONTAINER_LIMIT_PER_INTERVAL, 1);
     conf.setInt(OZONE_BLOCK_DELETING_LIMIT_PER_CONTAINER, 1);
     ContainerSet containerSet = new ContainerSet();
-    createToDeleteBlocks(containerSet, conf, 2, 1, 10);
+    int containerCount = 2;
+    int chunksPerBlock = 10;
+    int blocksPerContainer = 1;
+    createToDeleteBlocks(containerSet, conf, containerCount, blocksPerContainer,
+        chunksPerBlock);
 
     BlockDeletingServiceTestImpl service =
-        getBlockDeletinService(containerSet, conf, 1000);
+        getBlockDeletingService(containerSet, conf);
     service.start();
 
     try {
       GenericTestUtils.waitFor(service::isStarted, 100, 3000);
-      // 1st interval processes 1 container 1 block and 10 chunks
-      deleteAndWait(service, 1);
-      Assert.assertEquals(10, getNumberOfChunksInContainers(containerSet));
-
-      AtomicInteger timesToProcess = new AtomicInteger(1);
-      GenericTestUtils.waitFor(() -> {
-        try {
-          timesToProcess.incrementAndGet();
-          deleteAndWait(service, timesToProcess.get());
-          if (getNumberOfChunksInContainers(containerSet) == 0) {
-            return true;
-          }
-        } catch (Exception ignored) {}
-        return false;
-      }, 100, 100000);
-      Assert.assertEquals(0, getNumberOfChunksInContainers(containerSet));
```

Comment:
For tests in this class, I think it would be a good idea to still make an assertion on the number of chunks in the containers at the end of call.

Reply:
The problem with this test (and the service it tests) was that it used internal information about the layout.  Will check whether such information is still available without depending on filenames etc.